lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -342,16 +371,19 @@ void getGuesses(std::vector<Inst *> &Guesses,
 
     // if no empty slot, then push the guess to the result list
     if (CurrSlots.empty()) {
-      Guesses.push_back(JoinedGuess);
+      std::vector<Inst *> empty;
+      if (Prune(JoinedGuess, empty)) {
+        Guesses.push_back(JoinedGuess);
+      }
       continue;
     }
 
     // if there exist empty slots, then call getGuesses() recursively
     // and fill the empty slots
-    if (prune(JoinedGuess, CurrSlots)) {
+    if (Prune(JoinedGuess, CurrSlots)) {
```

Comment:
oops please leave functions in lowercase, variables in camelcase
https://llvm.org/docs/CodingStandards.html